Add security phase content (#9)
operate-first/SRE: sre_maturity.md

### Security - Phase 1: The ‘Crawl’ Phase

> Security are the controls (administrative, technical, physical) that are put in place to manage risk of a company to an acceptable level.

Reviewer: ```suggestion
Security is a set of controls (administrative, technical, physical) that are put in place to manage the risk of a company to an acceptable level.
```

> At this phase, access to the service is limited to those that strictly require it.

Reviewer: ```suggestion
In this phase, access to the service is limited to those that strictly require it.
```

Users that have access to the service have the minimum amount of permissions they need to operate the service.
Additionally, access is revoked as soon as possible when a user no longer needs to, for example when the user leaves the group / company.
All access to the system is logged.

> You have a mechanism for managing credentials & secrets, and delivering them as configuration into your services.

Reviewer: ```suggestion
A mechanism exists for managing credentials & secrets, and delivering them as configuration into your services.
```

> You are likely doing ad-hoc security activities such as assessments, vulnerability scans and pentesting.

Reviewer: ```suggestion
Engineers are likely doing ad-hoc security activities such as assessments, vulnerability scans, and pen-testing.
```

> These ad-hoc tests inform larger release milestones rather than continious releasing.

Reviewer: ```suggestion
These ad-hoc tests inform larger release milestones rather than continuous releases.
```

### Security - Phase 2: The 'Walk' Phase

In this phase you start to look at compliance.
Compliance relates to all the activities implemented to make sure that security controls meet requirements from compliance certifications/attestations (i.e. ISO 27001:2013, HIPAA, SOC 2 Type 2, PCI DSS, etc.).

> You may also have your own internal compliance processes or security standards depending on your company.

Reviewer: ```suggestion
Companies may also have your own internal compliance processes or security standards
```

Those compliance certifications/attestations are the tangible items you can give to your customers to provide assurances of your security posture.

> Regarding people, you implement background checks for individuals with privileged access to the environment.

Reviewer: ```suggestion
Background checks are implemented for individuals with privileged access to the environment.
```

> The more individuals with privileged access, the more background checks you have to run.

Reviewer: ```suggestion
The number of background checks needed grows linearly with the individuals with privileged access 
```

Security activities that were previously ad-hoc are now running regularly for each release, scanning your source & your production environment.
Doing 'security impact assessments' at the design phase of service features and architecture decisions pushes you security awareness earlier in the process.
This goes hand in hand with having a Common Vulnerabilities and Exposures (CVE) response process.

> All you dependecies are being scanned for vulneraiblities and you are responsive to vulnerability reports, able to push out fixes to production quickly in accordance with CVE severity.

Reviewer: ```suggestion
All code dependencies are being scanned for vulnerabilities and you are responsive to vulnerability reports and are able to push out fixes to production quickly in accordance with CVE severity.
```

### Security - Phase 1: The ‘Crawl’ Phase

Security are the controls (administrative, technical, physical) that are put in place to manage risk of a company to an acceptable level.

> At this phase you are not granular with authz policies to start with, but do have production access restricted to a tightly scoped set of individuals.

Reviewer 2: One of the thoughts that I have regarding this topic is there are two fundamental principles regarding authn and authz:

- Limited access: access to the service is limited to those that strictly require it.
- Minimum set of permissions: users that have access to the service have the minimum amount of permissions they need to operate the service.

Additionally:
- Access is revoked as soon as possible when a user no longer needs to. For example when the user leaves the group / company.
- All access to the system is logged.

IMO those are the fundamental pieces of security for crawl.

Reviewer 3: :100:

Granularity can come later with things like audited access, permissions escalation & tiered access as you mature the debugging experience of your service.
You should trust your team and assume positive intent.

> You have a mecahnismm for managing credentials & secrets, and deliverying them as configuration into your services.

Reviewer 2: ```suggestion
You have a mechanism for managing credentials & secrets, and delivering them as configuration into your services.
```

Additionally, you have a credential rotation policy that's based on time and individuals leaving your team.

You are likely doing ad-hoc security activities such as assessments, vulnerability scans and pentesting.
These ad-hoc tests inform larger release milestones rather than continious releasing.

### Security - Phase 2: The 'Walk' Phase

In this phase you start to look at compliance.
Compliance relates to all the activities implemented to make sure that security controls meet requirements from compliance certifications/attestations (i.e. ISO 27001:2013, HIPAA, SOC 2 Type 2, PCI DSS, etc.)

> Those compliance certifications/attestations are the tangible items you can give to your customers to provide assurances of your security posture.

Reviewer 2: It's important to include things like ESS, which is an internal compliance process.

Reviewer: It would be helpful to link to a public-facing doc on what ESS is. I'm not sure if it is well known outside of Red Hat

Author: I'm not aware of a public facing doc. It seems to be internally focused and driven only

Reviewer: In that case, unless we plan to publish it, what do you think about either skipping the reference or adding a few sentences clarifying that this is in an internal framework?

Author: Great.
ESS isn't mentioned anyways in the PR, so I'd prefer to keep it out.

This line hints at it, and should make the reader ask the question "Does my company have a process I should consider?"
`Companies may also have your own internal compliance processes or security standards`

Reviewer 3: Agreeing with @david-martin I think this is good to go.
